src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs

```diff
@@ -49,6 +50,43 @@ public static SocketError CreateSocket(AddressFamily addressFamily, SocketType s
             return socket.IsInvalid ? GetLastSocketError() : SocketError.Success;
         }
 
+        public static unsafe SocketError CreateSocket(
+            SocketInformation socketInformation,
+            out SafeSocketHandle socket,
+            ref AddressFamily addressFamily,
+            ref SocketType socketType,
+            ref ProtocolType protocolType)
+        {
+            if (socketInformation.ProtocolInformation == null || socketInformation.ProtocolInformation.Length < Interop.Winsock.WSAProtocolInfo.Size)
+            {
+                throw new ArgumentException(SR.net_sockets_invalid_socketinformation, nameof(socketInformation));
+            }
+
+            fixed (byte* pinnedBuffer = socketInformation.ProtocolInformation)
+            {
+                IntPtr handle = Interop.Winsock.WSASocketW(
+                    (AddressFamily)(-1),
+                    (SocketType)(-1),
+                    (ProtocolType)(-1),
+                    (IntPtr)pinnedBuffer, 0, Interop.Winsock.SocketConstructorFlags.WSA_FLAG_OVERLAPPED);
```

Comment:
I'd add a comment why WSA_FLAG_NO_HANDLE_INHERIT isn't set here.

Reply:
This is a good finding.

I'll be honest, this wasn't an intentional decision, I reused Framework code, where sockets were inheritable by default. I guess it's not possible to create inheritable duplicates of a non-inheritable handles but I will double-check.

Reply:
That was my assumption as well. I just noticed the discrepancy between this and the other calls to Interop.Winsock.WSASocketW across the repo: https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/Common/src/System/Net/SocketProtocolSupportPal.Windows.cs and https://github.com/dotnet/runtime/blob/4f9ae42d861fcb4be2fcd5d3d55d5f227d30e723/src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs

Reply:
I would now prefer to add `WSA_FLAG_NO_HANDLE_INHERIT` here for consistency, even if it has no effect.

Reply:
From what the docs say, WSA_FLAG_NO_HANDLE_INHERIT is used to make it so the socket isn't inheritable. My assumption was that if it was set, DuplicateSocket would fail. I'd try and confirm that.

Reply:
@jkotalik my understanding is that inheritance and duplication are different concepts. It's possible to `DuplicateHandle()` stuff into arbitrary unrelated processes, even if the handle is not inheritable, while handle inheritance is carried out in an automatic manner between parent and child processes.

Just made an experiment with sockets and it worked. However I definitely need to do some additional research to make sure my understanding is correct.

Reply:
Inheritable sockets allow child processes to access the socket; duplicate sockets allow any (non-child) process to take ownership of the socket.

Reply:
It's a good call out though; if the new default is to make all sockets non-inheritable (maybe not portable, perhaps some perf gain) then we should confirm if that's appropriate for these duplicated sockets too.